### PR TITLE
F5 policy conversion for F5 v11 config file

### PR DESCRIPTION
**Pull request by Rishabh-Baranwal**

### CODE REVIEW

File: python/avi/migrationtools/f5_converter/conversion_util.py [hunk 1]
```diff
@@ -604,12 +604,12 @@ def get_service_obj(self, destination, avi_config, enable_ssl,
...
 
 
-    def clone_pool(self, pool_name, vs_name, avi_pool_list, tenant=None):
+    def clone_pool(self, pool_name, clone_for, avi_pool_list, tenant=None):
         """
         If pool is shared with other VS pool is cloned for other VS as Avi dose not
         support shared pools with new pool name as <pool_name>-<vs_name>
         :param pool_name: Name of the pool to be cloned
-        :param vs_name: Name of the VS for pool to be cloned
+        :param clone_for: Name of the VS for pool to be cloned
```
Comment (grastogi23) [inline python/avi/migrationtools/f5_converter/conversion_util.py:618]: should this be clone_of_xxx

File: python/avi/migrationtools/f5_converter/conversion_util.py [hunk 2]
```diff
@@ -1496,19 +1498,37 @@ def vs_per_skipped_setting_for_references(self, avi_config):
...
-                                    pool_group_skipped_settings = \
-                                        self.get_pool_skipped_list(
-                                            avi_config, pool_group_name,
-                                            pool_csv_rows, csv_writer_dict_list,
-                                            vs_ref, profile_csv_list)
-                                    if pool_group_skipped_settings:
-                                        skipped_setting['Httppolicy'][
-                                            'Pool Group'] \
-                                            = pool_group_skipped_settings
+                                if http_req.get('switching_action',{}):
```
Comment (grastogi23) [inline python/avi/migrationtools/f5_converter/conversion_util.py]: let us split into another function. This is too long.

File: python/avi/migrationtools/f5_converter/conversion_util.py [hunk 3]
```diff
@@ -1788,6 +1808,108 @@ def set_pool_vrf(self, pool_ref, vrf_ref, avi_config):
...
+                                               avi_config['Pool'], tenant_name)
+                    if pool_ref:
+                        updated_pool_ref = self.get_object_ref(pool_ref,
+                                            conv_const.OBJECT_TYPE_POOL,
+                                            tenant_name, cloud_name)
+                        rule['switching_action']['pool_ref'] = updated_pool_ref
+        clone_policy['name'] += '-%s-clone' % vs_name
+        return clone_policy
+
+    def get_skipped_pool(self, avi_config, pool_name, pool_csv_rows,
```
Comment (grastogi23) [inline python/avi/migrationtools/f5_converter/conversion_util.py:1855]: add comments with high level logic

File: python/avi/migrationtools/f5_converter/conversion_util.py [hunk 4]
```diff
@@ -1788,6 +1808,108 @@ def set_pool_vrf(self, pool_ref, vrf_ref, avi_config):
...
+        }
+        pool_skipped_setting = {}
+        skipped_list = self.get_pool_skipped(pool_csv_rows, pool_name, vs_ref)
+        pool_object = [pool for pool in avi_config["Pool"]
+                       if pool['name'] == pool_name]
+        if skipped_list:
+            pool_skipped_setting['pool_name'] = pool_name
+            pool_skipped_setting['pool_skipped_list'] = skipped_list
+
+        if 'health_monitor_refs' in pool_object[0]:
```
Comment (grastogi23) [inline python/avi/migrationtools/f5_converter/conversion_util.py]: we should have a check for pool_object. If pool_object is empty [] then [0] will fail. Also, let us call it pool_objects or do pool_object = pool_object[0] if pool_object else None.

File: python/avi/migrationtools/f5_converter/conversion_util.py [hunk 5]
```diff
@@ -1788,6 +1808,108 @@ def set_pool_vrf(self, pool_ref, vrf_ref, avi_config):
...
+        :param prefix: clone for
+        :param avi_config: avi config dict
+        :param userprefix: prefix for objects
+        :return:None
+        """
+
+        policy_name = policy['name']
+        clone_policy = copy.deepcopy(policy)
+        for rule in clone_policy['http_request_policy']['rules']:
+            if rule.get('switching_action'):
```
Comment (grastogi23) [inline python/avi/migrationtools/f5_converter/conversion_util.py]: instead do if 'switching_action' in rule:

File: python/avi/migrationtools/f5_converter/vs_converter.py [hunk 6]
```diff
@@ -312,9 +314,77 @@ def convert_vs(self, vs_name, f5_vs, vs_state, avi_config, snat_config,
...
-                    vs_obj['http_policies'] = []
+                    if not vs_obj.get('http_policies'):
+                        vs_obj['http_policies'] = []
+                    else:
+                        ind = max([pol_index['index'] for pol_index in vs_obj[
+                                  'http_policies']])
+                        http_policies['index'] = ind + 1
                     vs_obj['http_policies'].append(http_policies)
                     avi_config['HTTPPolicySet'].append(policy)
+        if 'policies' in f5_vs:
```
Comment (grastogi23) [inline python/avi/migrationtools/f5_converter/vs_converter.py:325]: split this into a smaller function.

**Review by grastogi23 — CHANGES_REQUESTED** [review]

File: python/avi/migrationtools/f5_converter/conversion_util.py (hunk 1 above)
Comment (Rishabh-Baranwal) [reply python/avi/migrationtools/f5_converter/conversion_util.py:618]: It should be clone_for as we do cloning of pool/pool group for vs and policy

File: python/avi/migrationtools/f5_converter/conversion_util.py (hunk 2 above)
Comment (Rishabh-Baranwal) [reply python/avi/migrationtools/f5_converter/conversion_util.py]: changes as asked

File: python/avi/migrationtools/f5_converter/conversion_util.py (hunk 5 above)
Comment (Rishabh-Baranwal) [reply python/avi/migrationtools/f5_converter/conversion_util.py]: changed as asked

File: python/avi/migrationtools/f5_converter/conversion_util.py (hunk 3 above)
Comment (Rishabh-Baranwal) [reply python/avi/migrationtools/f5_converter/conversion_util.py:1855]: Added the doc string

File: python/avi/migrationtools/f5_converter/conversion_util.py (hunk 4 above)
Comment (Rishabh-Baranwal) [reply python/avi/migrationtools/f5_converter/conversion_util.py]: Added the pool_object check

File: python/avi/migrationtools/f5_converter/vs_converter.py (hunk 6 above)
Comment (Rishabh-Baranwal) [reply python/avi/migrationtools/f5_converter/vs_converter.py:325]: changed as asked

**Review by grastogi23 — APPROVED** [review]

**Pull request merged**